### ISSUE TITLE
function get_yaml_data enable error for warnings globally. Should use context manager

### ISSUE THREAD

**Issue by garthy** [wwkimball/yamlpath#40]
## Operating System

1. Name/Distribution: OracleLinux
2. Version: 7.7

## Version of Python and packages in use at the time of the issue.

1. [Distribution](https://wiki.python.org/moin/PythonDistributions):
2. Python Version: 3.6.8
3. Version of yamlpath installed: yamlpath==2.2.0
4. Version of ruamel.yaml installed: ruamel.yaml==0.16.5

## Minimum sample of YAML (or compatible) data necessary to trigger the issue
N/A
## Complete steps to reproduce the issue when triggered via:

import warnings
warnings.simplefilter("ignore")
#call get_yaml_data
get_yaml_data(.....)
# any warning should be ignored but exception is thrown
warnings.warn(......

## Expected Outcome

Warning ignored

## Actual Outcome

Exception thrown

## Screenshot(s), if Available

https://github.com/wwkimball/yamlpath/blob/d95b06ddbc86bc2d2c7e7259800789ffbf742134/yamlpath/func.py#L67

**Comment by wwkimball** [comment]
I will take a look at this issue.  The code you have cited is in place because the underlying processing library -- ruamel.yaml -- is excessively "noisy" during YAML parsing; it generates a vast amount of superfluous output.  I'll see if there's a way to detect whether `warnings.simplefilter("ignore")` is in effect before my code attempts to otherwise squelch ruamel.yaml's output.

Thanks for finding *and reporting* this!

**Comment by garthy** [comment]
You can simply put it in a `with         with warnings.catch_warnings():` context and do what ever you like. They way I'm working around this is

**Comment by garthy** [comment]
https://docs.python.org/3.1/library/warnings.html#temporarily-suppressing-warnings

**Comment by wwkimball** [comment]
I don't want to say "we can't do that" but I must say, "we can't solve this exactly like that".  If I suppress all warnings using that context manager as you demonstrate, it eats all warnings from ruamel.yaml without letting my code detect them.  This causes undesirable parsing side-effects.  For example:

```
---
# Duplicate aliases are expected to either:
# 1. Halt parsing at detection of the duplicate key (preferred because it prompts the user to clean their data, giving specific feedback as to where the problem alias is within the data)
# 2. Retain only the last-defined value for the duplicate alias (not preferred because it hides data quality issues that most often results in otherwise avoidable troubleshooting time)
aliases:
  - &duplicate_alias Original value
  - &duplicate_alias Override value

some_hash: &duplicate_alias
  child_key:  What exactly should *duplicate_alias be at this point?

some_key: *duplicate_alias
# or:
some_other_hash:
  <<: *duplicate_alias
```

I hope the reason we need to detect the `ReusedAnchorWarning` is evident from that data sample.  The only way to detect it is by trapping the warning during parsing.  If we suppress all warnings, parsing proceeds in an undefined state and users end up with garbled data or smaller -- harder to detect -- unexpected substitutions.

Let's look for another way to let you ignore all warnings without blocking yamlpath from intercepting warnings from ruamel.yaml.

**Comment by wwkimball** [comment]
@garthy Please test against the bugfix/40 branch -- https://github.com/wwkimball/yamlpath/tree/bugfix/40 -- if you are able.  The change works in my isolated testing but I'd like confirmation of the same within your specific use-case before I publish.

Thanks!!

**Comment by garthy** [comment]
I can confirm that works for me. 
Thanks

**Comment by wwkimball** [comment]
I have bundled this fix into today's release version 2.3.0, which also has expanded output formatting options for the yaml-paths command.  It should appear on package repositories very soon.

Thanks again for reporting this!
